frontend/src/pages/EditTeacher.vue

```diff
@@ -302,16 +211,49 @@ export default {
           this.getNativeLanguages();
           setTimeout(() => {
             this.$router.push({ path: `/teachers` });
-          }, 5000);
-        } else {
-          this.$buefy.notification.open({
+          }, 2000);
+      }
+      else if (value === false) {
+        this.$buefy.notification.open({
             message: "Something went wrong. Please try again.",
             duration: 3000,
             type: "is-warning",
             position: "is-top",
           });
-        }
-      });
+      }
+      this.resetTeacherUpdateSuccess(undefined)
```

Comment:
I would put this in the Vuex file, since you are already checking whether the PATCH a worked or not there (by status < 400). So you can dispatch the resetTeacherUpdateSuccess there instead.